src/System.Data.SqlClient/tests/FunctionalTests/DiagnosticTest.cs

```diff
@@ -0,0 +1,508 @@
+﻿using Xunit;
+using System.Reflection;
+using System.Diagnostics;
+using System.Collections;
+using System.Xml;
+using Xunit.Abstractions;
+using System.Threading.Tasks;
+
+namespace System.Data.SqlClient.Tests
+{
+    public class DiagnosticTest
+    {
+        private string _connectionString = "server=tcp:server,1432;database=test;uid=admin;pwd=SQLDB;connect timeout=60;";
+        private readonly ITestOutputHelper _output;
+
+        public DiagnosticTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        private void ExecuteScalarTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteScalar);
+        }
+        [Fact]
+        private void ExecuteScalarErrorTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteScalarError);
+        }
+        [Fact]
+        private void ExecuteNonQueryTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteNonQuery);
+        }
+        [Fact]
+        private void ExecuteNonQueryErrorTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteNonQueryError);
+        }
+        [Fact]
+        private void ExecuteReaderTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteReader);
+        }
+        [Fact]
+        private void ExecuteReaderErrorTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteReaderError);
+        }
+        [Fact]
+        private void ExecuteReaderWithCommandBehaviorTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteReaderWithCommandBehavior);
+        }
+        [Fact]
+        private void ExecuteXmlReaderTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteXmlReader);
+        }
+        [Fact]
+        private void ExecuteXmlReaderErrorTest()
+        {
+            CollectStatisticsDiagnostics(ExecuteXmlReaderError);
+        }
+
+        [Fact]
+        private void ExecuteScalarAsyncTest()
+        {
+            //await ExecuteScalarAsync();
```

Comment:
@nbilling I've removed this unnecessary comment and this change can be found in commit https://github.com/dotnet/corefx/pull/8420/commits/319736ba1954415602e7aecf25ba52daee275f36 Thanks! :smile: